json_object.c

```diff
@@ -720,22 +754,76 @@ int64_t json_object_get_int64(const struct json_object *jso)
 	}
 }
 
+uint64_t json_object_get_uint64(const struct json_object *jso)
+{
+	uint64_t cuint;
+
+	if (!jso)
+		return 0;
+	switch(jso->o_type)
+	{
+	case json_type_int:
+		switch(jso->o.c_int.cint_type) {
+		case json_object_int_type_int64:
+			if (jso->o.c_int.cint.c_int64 < 0)
+				return 0;
+			return (uint64_t)jso->o.c_int.cint.c_int64;
+		case json_object_int_type_uint64:
+			return jso->o.c_int.cint.c_uint64;
+		}
+	case json_type_double:
+		// UINT64_MAX can't be exactly represented as a double
+		// so cast to tell the compiler it's ok to round up.
+		if (jso->o.c_double >= (double)UINT64_MAX)
+			return UINT64_MAX;
+		if (jso->o.c_double < 0)
+			return 0;
+		return (uint64_t)jso->o.c_double;
+	case json_type_boolean:
+		return jso->o.c_boolean;
+	case json_type_string:
+		if (json_parse_uint64(get_string_component(jso), &cuint) == 0)
+			return cuint;
+		/* FALLTHRU */
+	default:
+		return 0;
+	}
+}
+
 int json_object_set_int64(struct json_object *jso,int64_t new_value){
-	if (!jso || jso->o_type!=json_type_int)
+	if (!jso || jso->o_type!=json_type_int || jso->o.c_int.cint_type != json_object_int_type_int64)
```

Comment:
As with _set_int(), this should set cint_type.
Also, let's avoid the dup code and have _set_int() call _set_int64(), since they do the exact same thing.

Reply:
ok, done